### PR TITLE
storage: Simplify fetches from storage

### PR DESCRIPTION
**Pull request by sboeuf**
The way we have been handling the fetch was too rigid and was forcing
us in doing several changes when adding new types of files.

With this commit, we rely on the magic of Go unmarshalling. The JSON
package handle all the complexity for us, allowing much more flexible
code.

### CODE REVIEW

**Comment by sameo** [comment]
LGTM

File: filesystem.go [hunk 1]
```diff
@@ -545,134 +555,55 @@ func (fs *filesystem) storeResource(podSpecific bool, podID, containerID string,
...
-		err = fs.fetchFile(path, &mounts)
-		if err != nil {
-			return nil, err
-		}
-
-		return mounts, nil
-
-	case devicesFileType:
-		devices := []Device{}
-		err = fs.fetchDeviceFile(path, &devices)
```
[the remark this answers is not in the document]
Comment (sboeuf) [reply filesystem.go]: Oh !! you're right! I have totally missed that...

[the remark this answers is not in the document]
Comment (sboeuf) [reply filesystem.go]: I'll fix that tomorrow first thing !

[the remark this answers is not in the document]
Comment (sboeuf) [reply filesystem.go]: Here it is https://github.com/containers/virtcontainers/pull/477

**Pull request merged**